compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDmaOpInterface.td

```diff
@@ -25,70 +25,233 @@ def DoublyStridedOpInterface : OpInterface<"DoublyStridedInterface"> {
+    This interface allows specifying mixed dynamic and static offsets, sizes and strides.
+
+    Ops that implement this interface need to expose the following methods:
+      1. `offsets`, `sizes` and `strides` variadic operands for both source and target.
+      2. `static_offsets`, resp. `static_sizes` and `static_strides` integer
+          array attributes for both source and target.
+
+    The invariants of this interface are:
+      1. `offsets`, `sizes` and `strides` have the same length.
+      2. if an entry of static access pattern operands is equal to a special sentinel value, 
+         namely `ShapedType::kDynamic`, then the corresponding entry is a dynamic value.
+      3. `offsets`, `sizes` and `strides` operands are non-negative.
   }];
   let cppNamespace = "mlir::iree_compiler::AMDAIE";
 
   let methods = [
     InterfaceMethod<
-      /*desc=*/"Returns ...",
-      /*retTy=*/"::mlir::ValueRange",
-      /*methodName=*/"getTargetOffsets"
+      /*desc=*/"Return the dynamic target offsets.",
+      /*retTy=*/"::mlir::OperandRange",
+      /*methodName=*/"getTargetOffsets",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getTargetOffsets();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return the dynamic target sizes.",
+      /*retTy=*/"::mlir::OperandRange",
+      /*methodName=*/"getTargetSizes",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getTargetSizes();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return the dynamic target strides.",
+      /*retTy=*/"::mlir::OperandRange",
+      /*methodName=*/"getTargetStrides",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getTargetStrides();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return the static target offsets.",
+      /*retTy=*/"::llvm::ArrayRef<int64_t>",
+      /*methodName=*/"getTargetStaticOffsets",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getTargetStaticOffsets();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return the static target sizes.",
+      /*retTy=*/"::llvm::ArrayRef<int64_t>",
+      /*methodName=*/"getTargetStaticSizes",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getTargetStaticSizes();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/"Return the static target strides.",
+      /*retTy=*/"::llvm::ArrayRef<int64_t>",
+      /*methodName=*/"getTargetStaticStrides",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getTargetStaticStrides();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Return a vector of all the static or dynamic target offsets of the op.
+      }],
+      /*retTy=*/"::llvm::SmallVector<::mlir::OpFoldResult, 4>",
```

Comment:
Nit: drop the `, 4` . What ends up happening is if you do something like

```
SmallVector<OpFoldResult, 5> ... = getTargetMixedOffsets();
```

then the move methods dont kick in and it creates a copy. So easiest is to just drop the size template parameter.

Reply:
Cool, thanks for pointing this out! Updated the PR.